### PR TITLE
Initial Extraction From Dolomite Engine 

### PR DESCRIPTION
**Pull request by fabianlim**
This is the initial extraction from the dolomite engine repo. 

Extracted models:
- `hf_models/models/gpt_dolomite`
- ~`hf_models/models/moe_dolomite`~

Conversion from HF supported
- `hf_models/model_conversion/bigcode`
- `hf_models/model_conversion/llama`
- ~`hf_models/model_conversion/mixtral`~

TODO:
- [x] adding CI (pypi, linting)
- remove some more unused code
   - [x] `modeling_utils/normalization/rmsnorm/torchtitan.py`
   - [x] `modeling_utils/normalization/rmsnorm/apex.py`
   - [x] `modeling_utils/normalization/layernorm/apex.py`
   - [x] `modeling_utils/normalization/layernorm/apex_persistent.py`
   - [x] `modeling_utils/embedding/ParameterizedEmbedding`
   - [x]  `modeling_utils/linear/ParameterizedLinear`
- adding some more notices
- [x] adding the conversion utilities.

### CODE REVIEW

**Comment by mayank31398** [comment]
@fabianlim I would also suggest dropping `ParametertizedEmbedding` and `ParametertizedLinear` and using the linear and embedding from torch directly
They are just for an experimental project I was working on.

**Comment by fabianlim** [comment]
@aldopareja this is more or less ok, but missing the notices. what do we want to put in the header of every file?

like this?
```
this code has been extracted from https://github.com/ibm-granite/dolomite-engine
```

**Comment by RobotSail** [comment]
We should also add the same publishing CI that we use elsewhere in instructlab so that it's easy to get stuff published.

**Comment by RobotSail** [comment]
This one https://github.com/instructlab/training/pull/31

**Comment by RobotSail** [comment]
> This one [instructlab/training#31](https://github.com/instructlab/training/pull/31)

Sorry not that one, this one: https://github.com/instructlab/training/pull/42

**Comment by RobotSail** [comment]
Nvm, I created a PR for publishing here, ignore the above comments: https://github.com/instructlab/GPTDolomite/pull/2

**Review by n1hility — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,7 +1,12 @@
 # IBM Dolomite Engine
 
-This repo was extracted from https://github.com/ibm-granite/dolomite-engine
-
-Original contributors:
+This repo contains code for IBM models, extracted from https://github.com/ibm-granite/dolomite-engine; all credit goes to the original authors:
 - mayank31398
 - kmehant
```
Comment (mayank31398) [inline README.md:5]: You can drop mehant, he is not an author

**Comment by mayank31398** [comment]
wow!!! 4k lines of code already :)

File: README.md (hunk 1 above)
Comment (fabianlim) [reply README.md]: done

**Comment by mayank31398** [comment]
Yikes!! all of my comments were ignored 🤣

**Comment by fabianlim** [comment]
@mayank31398 I thought you only gave these 2 comments
- removed the Parameterized version https://github.com/instructlab/GPTDolomite/pull/1/commits/0b7367b80d083dfe6fa94d1d2e03680ea983282c
- adjust the authorship https://github.com/instructlab/GPTDolomite/pull/1/commits/853c987b08d7ee6f7292cf8ed4354cdd2597c5f7

Was there anything else?

**Pull request merged**